SteamBot/Bot.cs

```diff
@@ -1084,6 +1085,74 @@ public void AcceptAllMobileTradeConfirmations()
             }                        
         }
 
+        /// <summary>
+        /// Get duration of escrow in days. Call this before sending a trade offer.
+        /// Credit to: https://www.reddit.com/r/SteamBot/comments/3w8j7c/code_getescrowduration_for_c/
+        /// </summary>
+        /// <param name="steamId">Steam ID of user you want to send a trade offer to</param>
+        /// <param name="token">User's trade token. Can be an empty string if user is on bot's friends list.</param>
+        /// <returns>TradeOfferEscrowDuration</returns>
+        public TradeOfferEscrowDuration GetEscrowDuration(SteamID steamId, string token)
+        {
+            var url = "https://steamcommunity.com/tradeoffer/new/";
+
+            var data = new System.Collections.Specialized.NameValueCollection();
+            data.Add("partner", steamId.AccountID.ToString());
+            if (!string.IsNullOrEmpty(token))
+            {
+                data.Add("token", token);
+            }            
+
+            var resp = SteamWeb.Fetch(url, "GET", data, false);
+            if (string.IsNullOrWhiteSpace(resp))
+            {
+                throw new NullReferenceException("Empty response from Steam when trying to retrieve escrow duration.");
+            }
+
+            return ParseEscrowResponse(resp);
+        }
+
+        /// <summary>
+        /// Get duration of escrow in days. Call this after receiving a trade offer.
+        /// </summary>
+        /// <param name="tradeOfferId">The ID of the trade offer</param>
+        /// <returns>TradeOfferEscrowDuration</returns>
+        public TradeOfferEscrowDuration GetEscrowDuration(string tradeOfferId)
+        {
+            var url = "http://steamcommunity.com/tradeoffer/" + tradeOfferId;
```

Comment:
If there was, there would be a solution for it.

Anyway tjhe problem mainly is that  "escrow_end_date" and k_ETradeOfferStateInEscrow only get set after trade accepted or declined :S.